presto-verifier/src/main/java/com/facebook/presto/verifier/rewrite/FunctionCallRewriter.java

```diff
@@ -0,0 +1,136 @@
+
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static java.util.Objects.requireNonNull;
+
+public class FunctionCallRewriter
+        extends DefaultTreeRewriter<Void>
+{
+    public static class FunctionCallSubstitute
+    {
+        private final QualifiedName name;
+        private final List<Integer> originalArgumentIndices;
+
+        public FunctionCallSubstitute(QualifiedName name, List<Integer> originalArgumentIndices)
+        {
+            this.name = requireNonNull(name, "name is null");
+            this.originalArgumentIndices = ImmutableList.copyOf(requireNonNull(originalArgumentIndices, "originalArgumentIndices is null"));
+        }
+
+        @Override
+        public boolean equals(Object o)
+        {
+            if (this == o) {
+                return true;
+            }
+            if (o == null || getClass() != o.getClass()) {
+                return false;
+            }
+            FunctionCallSubstitute that = (FunctionCallSubstitute) o;
+            return name.equals(that.name) && Objects.equals(originalArgumentIndices, that.originalArgumentIndices);
+        }
+
+        @Override
+        public int hashCode()
+        {
+            return Objects.hash(name, originalArgumentIndices);
+        }
+    }
+
+    private static final Pattern FUNCTION_CALL_SUBSTITUTION_PATTERN = Pattern.compile("/(\\w+)\\(([\\w|,]+)\\)/(\\w+)\\(([\\w|,]+)\\)/");
+
+    private final Map<QualifiedName, FunctionCallSubstitute> functionCallSubstituteMap;
+
+    private FunctionCallRewriter(Map<QualifiedName, FunctionCallSubstitute> functionCallSubstituteMap)
+    {
+        this.functionCallSubstituteMap = requireNonNull(functionCallSubstituteMap, "functionCallSubstituteMap is null.");
+    }
+
+    public static FunctionCallRewriter getInstance(String functionCallSubstitutes)
+    {
+        Map<QualifiedName, FunctionCallSubstitute> functionCallSubstituteMap = constructFunctionCallSubstituteMap(functionCallSubstitutes);
+        if (functionCallSubstituteMap == null || functionCallSubstituteMap.isEmpty()) {
+            return null;
+        }
+        return new FunctionCallRewriter(functionCallSubstituteMap);
+    }
+
+    public static Map<QualifiedName, FunctionCallSubstitute> constructFunctionCallSubstituteMap(String functionCallSubstitutes)
+    {
+        ImmutableMap.Builder<QualifiedName, FunctionCallSubstitute> map = ImmutableMap.builder();
+        if (functionCallSubstitutes == null) {
+            return map.build();
+        }
+
+        Matcher matcher = FUNCTION_CALL_SUBSTITUTION_PATTERN.matcher(functionCallSubstitutes);
+
+        while (matcher.find()) {
+            String originalName = matcher.group(1);
+            List<String> originalArgumentList = ImmutableList.copyOf(matcher.group(2).split(","));
+            String substituteName = matcher.group(3);
+            List<String> substituteArgumentList = ImmutableList.copyOf(matcher.group(4).split(","));
```

Comment:
This ".group(N)", is N a 1-based index?

Reply:
.group(1) refers to the group wrapped by the first paranthesis. .group(0) always refers to the entire match. Ex., for a(bc), group(0) returns abc, and group(1) return bc, if matched.

(Also got my memory refreshed by reading the java doc another time)